lib/improver/tests/nbhood/vicinity/test_ProbabilityOfOccurrence.py

```diff
@@ -0,0 +1,148 @@
+                                y_dimension_values=y_dimension_values,
+                                x_dimension_values=y_dimension_values)
+
+    def test_with_realization(self):
+        """Test when a realization coordinate is present."""
+        expected = np.array(
+            [[1., 1., 0.83333333, 0.66666667, 0.5],
+             [0.66666667, 0.77777778, 0.77777778, 0.77777778, 0.66666667],
+             [0.33333333, 0.55555556, 0.77777778, 1., 1.],
+             [0., 0.22222222, 0.44444444, 0.66666667,  0.66666667],
+             [0., 0.16666667, 0.33333333, 0.5, 0.5]])
+        distance = 2000
+        neighbourhood_method = "square"
+        radii = 2000
+        result = (
+            ProbabilityOfOccurrence(
+                distance, neighbourhood_method, radii).process(self.cube))
+        self.assertIsInstance(result, Cube)
+        self.assertArrayAlmostEqual(result.data, expected)
+
+    def test_without_realization(self):
+        """Test when a realization coordinate is not present."""
+        expected = np.array(
+            [[1., 1., 0.83333333, 0.66666667, 0.5],
+             [0.66666667, 0.77777778, 0.77777778, 0.77777778, 0.66666667],
+             [0.33333333, 0.55555556, 0.77777778, 1., 1.],
+             [0., 0.22222222, 0.44444444, 0.66666667,  0.66666667],
+             [0., 0.16666667, 0.33333333, 0.5, 0.5]])
+        cube = self.cube[0, :, :, :]
+        cube.remove_coord("realization")
+        distance = 2000
+        neighbourhood_method = "square"
+        radii = 2000
+        result = (
+            ProbabilityOfOccurrence(
+                distance, neighbourhood_method, radii).process(self.cube))
+        self.assertIsInstance(result, Cube)
+        self.assertArrayAlmostEqual(result.data, expected)
+
+    def test_additional_arguments(self):
+        """Test when all keyword arguments are passed in."""
+        expected = np.array(
+            [[[1., 1., 0.8, 0.4, 0.2],
+              [0.8, 0.8, 1., 0.8, 0.8],
+              [0.2, 0.4, 0.8, 1., 1.],
+              [0., 0.2, 0.6, 0.8, 0.8],
+              [0., 0., 0.2, 0.2, 0.2]],
+             [[0.92307692, 0.84615385, 0.76923077, 0.53846154, 0.38461538],
+              [0.69230769, 0.76923077, 0.84615385, 0.76923077, 0.69230769],
+              [0.38461538, 0.53846154, 0.69230769, 0.76923077, 0.84615385],
+              [0.15384615, 0.30769231, 0.46153846, 0.61538462, 0.69230769],
+              [0., 0.07692308, 0.23076923, 0.30769231, 0.30769231]]])
+        data = np.zeros((1, 2, 5, 5))
+        data[0, :, 0, 1] = 1.0
+        data[0, :, 2, 3] = 1.0
+        y_dimension_values = np.arange(0.0, 10000.0, 2000.0)
+        cube = set_up_cube(data, "lwe_precipitation_rate", "m s-1",
+                           y_dimension_values=y_dimension_values,
+                           x_dimension_values=y_dimension_values,
+                           timesteps=np.array([402192.5, 402195.5]))
+        distance = 2000
+        neighbourhood_method = "circular"
```

Comment:
The acceptance criteria only request a square neighbourhood, so I believe anything to do with the circular neighbourhood is really out of scope. I'm entirely open to being corrected.